propagators/opentelemetry-propagator-instana/src/InstanaPropagator.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Context,
+  trace,
+  isSpanContextValid,
+  isValidSpanId,
+  isValidTraceId,
+  TextMapGetter,
+  TextMapPropagator,
+  TextMapSetter,
+  TraceFlags,
+  INVALID_TRACEID,
+  INVALID_SPANID,
+} from '@opentelemetry/api';
+
+/** Instana header names */
+export const INSTANA_TRACE_ID_HEADER = 'X-INSTANA-T';
+export const INSTANA_SPAN_ID_HEADER = 'X-INSTANA-S';
+export const INSTANA_LEVEL_HEADER = 'X-INSTANA-L';
+
+const FIELDS = [
+  INSTANA_TRACE_ID_HEADER,
+  INSTANA_SPAN_ID_HEADER,
+  INSTANA_LEVEL_HEADER,
+];
+
+function readHeader(
+  carrier: unknown,
+  getter: TextMapGetter,
+  key: string
+): string {
+  let header = getter.get(carrier, key.toLowerCase());
+  if (Array.isArray(header)) [header] = header;
+  return header || '';
```

Comment:
Why not return `undefined` here as this would allow to shorten code below form `traceId && traceId.length < 32` to `traceId?.length < 32`.